packages/file_selector/file_selector_android/android/src/main/java/io/flutter/plugins/file_selector/FileSelectorPlugin.java

```diff
@@ -99,24 +104,41 @@ private void tearDown() {
     }
   }
 
+  @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
   @Override
   public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result rawResult) {
     if (activityState == null || activityState.getActivity() == null) {
       rawResult.error("no_activity", "file_selector plugin requires a foreground activity.", null);
       return;
     }
+
+    final HashMap arguments = (HashMap) call.arguments;
+
     MethodChannel.Result result = new MethodResultWrapper(rawResult);
     FileSelectorDelegate delegate = activityState.getDelegate();
+    delegate.clearCache();
     switch (call.method) {
       case METHOD_GET_DIRECTORY_PATH:
         delegate.getDirectoryPath(call, result);
         break;
       case METHOD_GET_SAVE_PATH:
         throw new UnsupportedOperationException("getSavePath is not supported yet");
       case METHOD_OPEN_FILE:
-        throw new UnsupportedOperationException("openFile is not supported yet");
+        String[] mimeTypes = getMimeTypes(arguments);
+        boolean isMultipleSelection = (boolean) arguments.get("multiple");
+        delegate.openFile(call, result, isMultipleSelection, mimeTypes);
+        break;
       default:
         throw new IllegalArgumentException("Unknown method " + call.method);
     }
   }
+
+  @VisibleForTesting
+  private String[] getMimeTypes(HashMap arguments) {
+    ArrayList acceptedTypeGroups = (ArrayList) arguments.get("acceptedTypeGroups");
+    HashMap xTypeGroups = (HashMap) acceptedTypeGroups.get(0);
+    ArrayList<String> mimeTypesList = (ArrayList<String>) xTypeGroups.get("mimeTypes");
+
+    return mimeTypesList.toArray(new String[0]);
```

Comment:
what is the purpose of this?

Reply:
Since we need to pass the mimeTypes to filter to the intent, this converts the arguments to an array.